### ISSUE TITLE
Could not resolve IPv6 address of host

### ISSUE THREAD

**Issue by zhengqwe** [curl/curl#12134]
### I did this

I am sure that `example.com` has IPv6 address

    curl https://example.com -6 --trace -
    == Info: Could not resolve host: example.com
    == Info: Closing connection
    curl: (6) Could not resolve host: example.com

### I expected the following

resolve IPv6 address normally

### curl/libcurl version

curl 8.4.0 (x86_64-w64-mingw32) libcurl/8.4.0 OpenSSL/3.1.3 (Schannel) zlib/1.3 brotli/1.1.0 zstd/1.5.5 WinIDN libssh2/1.11.0 nghttp2/1.57.0 ngtcp2/0.19.1 nghttp3/0.15.0
Release-Date: 2023-10-11
Protocols: dict file ftp ftps gopher gophers http https imap imaps ldap ldaps mqtt pop3 pop3s rtsp scp sftp smb smbs smtp smtps telnet tftp ws wss
Features: alt-svc AsynchDNS brotli HSTS HTTP2 HTTP3 HTTPS-proxy IDN IPv6 Kerberos Largefile libz MultiSSL NTLM SPNEGO SSL SSPI threadsafe UnixSockets zstd

### operating system

Windows 10

**Comment by gvanem** [comment]
Your curl is not using C-ares. So perhaps an issue with `getaddrinfo()`.
How does `ping -6 example.com`  work for you?

**Comment by bagder** [comment]
I cannot reproduce on Linux at least

~~~
$ curl -s6 example.com -o /dev/null -v
*   Trying [2606:2800:220:1:248:1893:25c8:1946]:80...
* Connected to example.com (2606:2800:220:1:248:1893:25c8:1946) port 80
> GET / HTTP/1.1
> Host: example.com
> User-Agent: curl/8.3.0
> Accept: */*
...

**Comment by vszakats** [comment]
I can confirm with official builds starting with the initial 8.3.0 release (8.3.0_1).

Reverting d3142b57299f3aeec546cc1179b9ff170f8d20ad fixed it for me.
Test binary here: https://github.com/curl/curl-for-win/suites/17265612315/artifacts/986334607

**Comment by bagder** [comment]
Wow. Now on which Windows versions do you see this? It smells like a `getaddrinfo` bug?

**Comment by vszakats** [comment]
This was on a Win7 machine.

**Comment by bagder** [comment]
#12136 feels terrible lame but should fix this

**Comment by vszakats** [comment]
@zhengqwe Can you re-run your test with the test curl.exe linked above?

**Comment by vszakats** [comment]
Test binary here attempting to fix this with #12217:
https://ci.appveyor.com/api/buildjobs/tmqd8tt2cu25snjt/artifacts/all-mingw-8.4.0_5.zip

Can you verify if this build fixes the issue?
